package/expo-package/src/handlers/Audio.ts

```diff
@@ -212,13 +212,39 @@ export type RecordingOptions = {
   keepAudioActiveHint?: boolean;
 };
 
+const sleep = (ms: number) =>
+  new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, ms);
+  });
+
 export const Audio = AudioComponent
-  ? {
-      startRecording: async (recordingOptions: RecordingOptions, onRecordingStatusUpdate) => {
+  ? class {
+      recording: typeof RecordingObject | null;
+
+      constructor() {
+        this.recording = null;
+      }
+      async startRecording(recordingOptions: RecordingOptions, onRecordingStatusUpdate) {
```

Comment:
it would be good if you stick to arrow functions, as otherwise one can mess up with the `this` pointer.

```ts
startRecording = async (recordingOptions: RecordingOptions, onRecordingStateUodate: [WhatIsTheTypeHere]) => {
....
}
```

Reply:
True - thanks !